src/pykeen/nn/representation.py

```diff
@@ -1426,3 +1427,111 @@ def __init__(
         assignment[mask, 1] = torch.arange(backfill.max_id)
 
         super().__init__(assignment=assignment, bases=[base, backfill], **kwargs)
+
+
+class TransformedRepresentation(Representation):
+    """
+    A (learnable) transformation upon base representations.
+
+    In the following example, we create representations which are obtained from a trainable transformation of fixed
+    random walk encoding features. We first load the dataset, here Nations:
+
+    >>> from pykeen.datasets import get_dataset
+    >>> dataset = get_dataset(dataset="nations")
+
+    Next, we create a random-walk positional encoding of dimension 32:
+
+    >>> from pykeen.nn import init
+    >>> dim = 32
+    >>> initializer = RandomWalkPositionalEncoding(triples_factory=dataset.training)
```

Comment:
this should apparently be 
```python
initializer = init.RandomWalkPositionalEncoding(triples_factory=dataset.training, dim=dim)
```

However, after executing this example, the shape of the initialized tensor is always `dim-1`, eg, 31 in this case, or 63 if ask for `dim=64` 🤔

Reply:
Actually, this actual `dim-1` dimension would also clash with the defined MLP encoder and TransformedRepresentation as they'd expect, eg, a 32d input but we'd get only 31d

Reply:
can you propose the update to go along with this?

Reply:
yes, fixed in https://github.com/pykeen/pykeen/pull/984/commits/8cc05bad61c409c5d43ed76cb477968d3cff092b